text/0005-syncing.md

```diff
@@ -0,0 +1,114 @@
+
+Same mechanism can be used in a node that is catching up – it will use `Block`s from the linear chain to build up its Global State via `BlockExecutor` (we need a mechanism of turning `Block` into `FinalizedBlock` as that's the input that `BlockExecutor` understands) and create new eras. 
+
+Obviously, over the course of synchronization some eras will become obsolete but the assumption is that eras are deactivated/removed by a separate mechanism. For example, `BlockExecutor` component already knows about the unbonding delay so it could send an event to a consensus component that it should deactivate an era (or if consensus component needs this knowledge it could deactivate old eras by itself).
+
+#### Variant B ####
+
+Since downloading the whole linear chain (and most importantly, the deploys) is a time-consuming operation, a faster alternative would be to just sync the Global State at the initial hash. Global State is a merkle trie so it's easily splittable and has an easy way of verifying its correctness (compare root hashes).
+
+If we choose to synchronize Global State directly though, the above won't work. We need to be able to create active eras just by looking at the Global State. This should also be fairly trivial since PoS contract can track past eras, current ones and future (auctions already executed but era not yet started). We can use information from the cache and create instances of eras that are still active.
+
+**Necessary work:**
+
+1. Start a node with the trusted hash
+2. Ask peers for the initial block from the linear chain.
+3. Synchronize Global State at the post-state hash of the initial block.
+4. Validate correctness of the Global State with the post-state hash from the initial block.
+
+
+---------------------
+Once whole linear chain is "consumed" and state is built-up, we need to synchronize DAG(s) of still active eras.
+
+### Synchronizing DAG(s) ###
+
+#### Variant A (quick-n-dirty) ####
+
+Once node has built up its Global State and has necessary eras started it could rely on already existing mechanism of syncing nodes – i.e. it could transition into an *active* node (not a validator) and wait for new messages. Upon receiving a message for which it doesn't have all the necessary dependencies it will already start synchronizing.
+
+#### Variant B
+
+If we want to fully sync the DAG(s) before transitioning from the synchronizing to an active node modes we need to ask peers for their latest DAG tips and compare it to our own view. Requires:
+
+* adding a new variant to `HighwayMessage::GetTips` which should return node's `panorama`
+* joining node should send `GetTips` message to its peers for all active eras
```

Comment:
I'm wondering whether instead of trusting a random selection of peers, we could download all dependencies of the switch block.
For that, we'd ideally know the hash of the era's last _Highway block_, thought… not sure if we want to leak that into e.g. the first executed block of the next era or something…

Reply:
The main reason to send `GetTips` to multilpe nodes is to parallelize the synchronization process. Node should still add only correct vertices to the DAG (since it already knows validators for active eras).